src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -538,11 +538,93 @@ void FastRouteCore::applyHorizontalAdjustments(const odb::Point& first_tile,
   }
 }
 
-int FastRouteCore::getEdgeCapacity(int x1,
-                                   int y1,
-                                   int x2,
-                                   int y2,
-                                   int layer)
+void FastRouteCore::addVerticalAdjustments(
+    const odb::Point& first_tile,
+    const odb::Point& last_tile,
+    int layer,
+    interval<int>::type first_tile_reduce_interval,
+    interval<int>::type last_tile_reduce_interval)
+{
+  // add intervals to set for each tile
+  for (int x = first_tile.getX(); x <= last_tile.getX(); x++) {
+    for (int y = first_tile.getY(); y < last_tile.getY(); y++) {
+      if (x == first_tile.getX()) {
+        vertical_blocked_intervals_[std::make_tuple(x, y, layer)]
+            += first_tile_reduce_interval;
+      } else if (x == last_tile.getX()) {
+        vertical_blocked_intervals_[std::make_tuple(x, y, layer)]
+            += last_tile_reduce_interval;
+      } else {
+        addAdjustment(x, y, x, y + 1, layer, 0, true);
+      }
```

Comment:
why are we only tracking intervals in the first & last tiles?

Reply:
@luis201420 could you comment here?

Reply:
Because the other (middle) tiles I am reducing to 0 because they are crossed by the block completely.